docs/upgrading-to-datastreams.asciidoc

```diff
@@ -0,0 +1,98 @@
+
+Users that are not ready to <<upgrade-to-apm-integration>> should
+upgrade to data streams in the meantime.
+
+[discrete]
+[[why-upgrade-to-data-streams]]
+=== Why upgrade?
+
+Data streams offer a number of benefits over classic indices:
+
+**Data streams**:
+
+* Reduced number of fields per index, better space efficiency, and faster queries
+* More granular data control
+* Errors and metrics data streams are shared with other data sources -- which means better long-term integration with the logs and metrics apps
+* Removes template inheritance for ILM policies and makes use of new {es} index and component templates
+* Fixes <<server-resource-exists-not-alias>>
+
+[discrete]
+[[considerations-data-streams]]
+=== Data streams considerations
+
+* This change cannot be reverted
+* Users on {ece} require additional steps prior to migrating, like configuring TLS certificates for the connection between APM Server and {es}
+
+[discrete]
+[[apm-data-streams-upgrade-steps]]
+=== Upgrade a self-managed APM Server
+
+. <<apm-data-streams-upgrade-1>>
+. <<apm-data-streams-upgrade-2>>
+. <<apm-data-streams-upgrade-3>>
+. <<apm-data-streams-upgrade-4>>
+
+[discrete]
+[[apm-data-streams-upgrade-1]]
+==== Upgrade the Elastic Stack
+
+The Elastic Stack ({es} and {kib}) must be upgraded prior to APM Server.
+See the {stack-ref}/upgrading-elastic-stack.html[Elastic Stack Installation and Upgrade Guide] for guidance.
+
+Review the APM <<release-notes,release notes>>, <<apm-breaking,breaking changes>>,
+and Observability {observability-guide}/whats-new.html[What's new] content for important changes between
+your current APM version and this one.
+
+[discrete]
+[[apm-data-streams-upgrade-2]]
+==== Add the APM integration
+
+Before enabling data streams in APM Server, the APM integration must be added to your cluster.
+This enables Fleet to set up ingest templates, ILM policies,
+and ingest pipelines when APM Server tries to connect.
+
+. In {kib}, navigate to **Management** > **Fleet**.
+First time? The page may take a few minutes to load.
+
+. Select **Agent policies** > **Default policy** and click **Add integration**.
+
+. Search for and select **Elastic APM**. Under step one, select **APM integration**.
+
+. Click **Add Elastic APM**. Configuring the integration is not required as we're only using the integration to set up {es} resources. Click **Save and Continue** and then **Add {agent} later** (you don't need to add an {agent} later).
+
+[discrete]
+[[apm-data-streams-upgrade-3]]
+==== Configure APM Server to use data streams
+
+Stop APM Server and set <<_configuration_options_data_streams,`data_streams.enabled: true`>> in apm-server.yml:
+
+[source,yaml]
+----
+apm-server:
+  data_streams.enabled: true
+----
+
+<<apm-server-starting,Restart APM Server>>.
+
+[discrete]
+[[apm-data-streams-upgrade-4]]
+==== Configure resources
```

Comment:
ILM & Ingest pipelines. What else am I missing here?

Reply:
I think this covers it -- users need to set their own ILM policies and ingest pipelines for the datastreams. Is there anything else @simitt?

Reply:
Why do they _need_ to configure this (rather they _can_ customize if they want to)? Data streams should still be set up with default policies and ingest pipelines.

Reply:
Good point! Fixing...